### PR TITLE
Task 22 - Refactoring Url Entity class

### PR DESCRIPTION
**Pull request by BartoszKarp**

### CODE REVIEW

File: src/main/java/pl/stormit/tinyurl/domain/model/Url.java [hunk 1]
```diff
@@ -5,23 +5,16 @@
 
 import javax.persistence.*;
 import java.time.LocalDate;
+import java.util.UUID;
 
 @Entity
 @Table(name = "urls")
 @ToString
```
Comment (RobertoJavaDev) [inline src/main/java/pl/stormit/tinyurl/domain/model/Url.java:12]: Maybe use Lombok annotation without getter and setter method in class

File: src/main/java/pl/stormit/tinyurl/domain/model/Url.java [hunk 2]
```diff
@@ -5,23 +5,16 @@
...
-            name = "url_sequence",
-            sequenceName = "url_sequence",
-            allocationSize = 1
-    )
-    @GeneratedValue(
-            strategy = GenerationType.SEQUENCE,
-            generator = "url_sequence"
-    )
-    private Long id;
+    private UUID id;
```
[the remark this answers is not in the document]
Comment (BartoszKarp) [reply src/main/java/pl/stormit/tinyurl/domain/model/Url.java:17]: id is generating and application is working

File: src/main/java/pl/stormit/tinyurl/domain/model/Url.java (hunk 1 above)
Comment (BartoszKarp) [reply src/main/java/pl/stormit/tinyurl/domain/model/Url.java:12]: I don't think it's needed. In all constructors I must add id generator:

public Url() {
        this.id = UUID.randomUUID();
    }

Comment (BartoszKarp) [reply src/main/java/pl/stormit/tinyurl/domain/model/Url.java:12]: OK now I understand what do you mean :)

File: src/main/java/pl/stormit/tinyurl/domain/model/Url.java [hunk 3]
```diff
@@ -5,23 +5,16 @@
 
 import javax.persistence.*;
 import java.time.LocalDate;
+import java.util.UUID;
```
[the remark this answers is not in the document]
Comment (RobertoJavaDev) [reply src/main/java/pl/stormit/tinyurl/domain/model/Url.java:11]: The reason why we should use javax.validation.constraints.NotNull is that this annotation is implemented by Hibernate validator.

com.sun.istack - the annotation is an internal class, it's not for public use.
Even if the class is in the classpath, it doesn't necessarily mean that you can safely refer to that class. 
Oracle discourages developers from using classes from such packages.

**Review by RobertoJavaDev — APPROVED** [review]
Looks good to me

**Pull request merged**